File: analyzers/tests/SonarAnalyzer.Test/TestCases/UseCharOverloadOfStringMethods.Framework.vb
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Linq

Class Testcases

    Sub Simple()
        Dim str = "hello"

        ' "char" overloads do not exist on .NET Framework
        str.StartsWith("x") ' Compliant
        str.EndsWith("x") ' Compliant

        str.StartsWith("x"c) ' Compliant
        str.EndsWith("x"c) ' Compliant
    End Sub

End Class

File: analyzers/tests/SonarAnalyzer.Test/TestCases/UseCharOverloadOfStringMethods.Framework.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;

class Testcases
{
    void Simple()
    {
        var str = "hello";

        // "char" overloads do not exist on .NET Framework
        str.StartsWith("x");    // Compliant
        str.EndsWith("x");      // Compliant

        str.StartsWith('x');    // Error [CS1503]
        str.EndsWith('x');      // Error [CS1503]
    }
}
